Common/Product/SharedProject/CommonUtils.cs

```diff
@@ -240,28 +180,28 @@ public static string GetAbsoluteDirectoryPath(string root, string relativePath)
         /// either path is invalid.</exception>
         public static string GetAbsoluteFilePath(string root, string relativePath)
         {
-            var rootUri = MakeUri(root, true, UriKind.Absolute, "root");
-            var relUri = MakeUri(relativePath, false, UriKind.RelativeOrAbsolute, "relativePath");
+            var absolutePath = HasStartSeparator(relativePath)
+                ? Path.GetFullPath(relativePath)
+                : Path.Combine(root, relativePath);
 
-            Uri absUri;
+            var split = absolutePath.Split(DirectorySeparators);
+            var segments = new List<string>();
 
-            if (relUri.IsAbsoluteUri)
-            {
-                absUri = relUri;
-            }
-            else
+            for (var i = split.Length - 1; i >= 0; i--)
             {
-                try
+                var segment = split[i];
+
+                if (segment == "..")
                 {
-                    absUri = new Uri(rootUri, relUri);
+                    i--;
                 }
-                catch (UriFormatException ex)
+                else if(segment != ".")
                 {
-                    throw new InvalidOperationException("Cannot create absolute path", ex);
+                    segments.Insert(0, segment);
```

Comment:
Consider either switching to a linked list or building the list backwards and then reversing it.

Reply:
Good catch, I wasn't aware that the Insert is an O(n), I didn't expect that. LinkedList makes more sense. Fixed.